src/PatchRequest.php

```diff
@@ -45,7 +48,7 @@ public function setStrategyCommand(): SetStrategy
         Assert::keyExists($this->requestData, 'strategy_type');
         Assert::string($this->requestData['strategy_id']);
         Assert::string($this->requestData['strategy_type']);
-        /** @var array<array<string, mixed>>|null $segments */
+        /** @var array<array<string, array<string, mixed>|string>>|null $segments */
```

Comment:
I'm not sure if we could define the following "Object-like array" type for the `segments` array:
```suggestion
        /** @var array<array{segment_id: string, segment_type: string, criteria: array<string, mixed>}>|null $segments */
```

Reply:
In this case, we are in the same as the explained when trying to force the "WriteSegment", if we put a more strict type here, Psalm will fail because it understands that the assertions are redundant. 
![image](https://user-images.githubusercontent.com/1093654/142756969-4e43f299-5cf2-434e-a01d-d8aca932b594.png)